s2spy/_base_calendar.py

```diff
@@ -344,47 +345,50 @@ def __repr__(self) -> str:
     def visualize(
         self,
         n_years: int = 3,
+        interactive: bool = False,
         relative_dates: bool = False,
         add_length: bool = False,
     ) -> None:
         """Plots a visualization of the current calendar setup, to aid in user setup.
 
+        Note: The interactive visualization requires the `bokeh` package to be installed
+        in the active Python environment.
+
         Args:
-            add_length: Toggles if the frequency of the intervals should be displayed.
-                      Defaults to False (Matplotlib plotter only)
             n_years: Sets the maximum number of anchor years that should be shown. By
                      default only the most recent 3 are visualized, to ensure that they
                      fit within the plot.
+            interactive: If False, matplotlib will be used for the visualization. If
+                         True, bokeh will be used.
+            relative_dates: Toggles if the intervals should be displayed relative to the
+                            anchor date, or as absolute dates.
+            add_length: Toggles if the frequency of the intervals should be displayed.
+                      Defaults to False (Matplotlib plotter only)
         """
-        n_years = max(n_years, 1)
-        n_years = min(n_years, len(self.get_intervals().index))
-        _plot.matplotlib_visualization(self, n_years, relative_dates, add_length)
-
-    def visualize_interactive(
-        self,
-        relative_dates: bool,
-        n_years: int = 3,
-    ) -> None:
-        """Plots a visualization of the current calendar setup using `bokeh`.
+        calendar = copy.deepcopy(self)
+        mapped = calendar._mapping is not None  # pylint: disable=protected-access
+        if not mapped:
+            calendar.map_years(2000, 2000)
+            if not relative_dates:
+                print("Setting relative dates to True, as the calendar is not mapped yet.")
```

Comment:
"calendar is not mapped yet" may be unclear. Perhaps something like: "Setting relative_dates=True. To see absolute dates, first call calendar.map_years or calendar.map_data."